src/constructors.ts

```diff
@@ -110,28 +110,34 @@ function fromSuccess<O, P extends any[]>(
  * ```
  */
 function applySchema<ParsedInput, ParsedContext>(
-  inputSchema?: ParserSchema<ParsedInput>,
-  contextSchema?: ParserSchema<ParsedContext>,
+  inputSchema?: StandardSchemaV1<ParsedInput>,
+  contextSchema?: StandardSchemaV1<ParsedContext>,
 ) {
   return <R, Input extends ParsedInput, Context extends ParsedContext>(
     fn: (input: Input, context: Context) => R,
   ): ApplySchemaReturn<ParsedInput, ParsedContext, typeof fn> => {
     const callable = ((input?: unknown, context?: unknown) => {
-      const ctxResult = (contextSchema ?? alwaysUnknownSchema).safeParse(
-        context,
+      const ctxResult = (contextSchema ?? alwaysUnknownSchema)['~standard']
+        .validate(
+          context,
+        )
+      const result = (inputSchema ?? alwaysUnknownSchema)['~standard'].validate(
+        input,
       )
-      const result = (inputSchema ?? alwaysUnknownSchema).safeParse(input)
+      if (result instanceof Promise || ctxResult instanceof Promise) {
+        throw new TypeError('Schema validation must be synchronous')
+      }
```

Comment:
We only allow sync validations at the moment but standard schema makes it easy to accept async.
Should we allow it?
Remember that the reasoning to remove it was that any async validation should go within the body of the function but allowing async validations will make reuse of schemas easier.

Reply:
I would allow to avoid the branch with the exception. I still believe it is a sort of anti-function. But having this exception here hurts more